### PR TITLE
Refactor prepare-machine.ps1

### PR DESCRIPTION
**Pull request by nibanks**
## Description

Cleans up the prepare-machine.ps1 script and its usages.

## Testing

Leveraging the CI.

### CODE REVIEW

File: .azure/templates/run-bvt.yml [hunk 1]
```diff
@@ -42,14 +41,14 @@ jobs:
     inputs:
       pwsh: true
       filePath: scripts/prepare-machine.ps1
-      ${{ if and(eq(parameters.kernel, true), eq(parameters.testCerts, true)) }}:
```
Comment (nibanks) [inline .azure/templates/run-bvt.yml:45]: Let's refactor to use the `extraPrepareArgs` model.

File: .github/workflows/check-clog.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5
     - name: Prepare Machine
-      run: scripts/prepare-machine.ps1 -Configuration Dev
+      run: scripts/prepare-machine.ps1
```
Comment (nibanks) [inline .github/workflows/check-clog.yml:33]: Do we really need build and test?

File: scripts/prepare-machine.ps1 [hunk 3]
```diff
@@ -4,147 +4,167 @@
...
-$RootDir = Split-Path $PSScriptRoot -Parent
-$NuGetPath = Join-Path $RootDir "nuget"
-
-
-$MessagesAtEnd = New-Object Collections.Generic.List[string]
-
 if ($PSVersionTable.PSVersion.Major -lt 7) {
-
+    # This script requires PowerShell core (mostly for xplat stuff).
     Write-Error ("`nPowerShell v7.x is needed for this script to work. " +
```
Comment (maolson-msft) [inline scripts/prepare-machine.ps1:96]:
```suggestion
    Write-Error ("`nPowerShell v7.x or greater is needed for this script to work. " +
```

File: scripts/prepare-machine.ps1 [hunk 4]
```diff
@@ -4,147 +4,167 @@
...
         Expand-Archive -Path $ZipPath -DestinationPath $ArtifactsPath -Force
         Remove-Item -Path $ZipPath
     }
 }
 
-function Download-Xdp-Kit {
-    if ($Force) { rm -Force -Recurse $ArtifactsPath }
-    if (!(Test-Path $ArtifactsPath)) { mkdir $ArtifactsPath }
+# Downloads the latest version of XDP (for building).
+function Install-Xdp {
```
Comment (maolson-msft) [inline scripts/prepare-machine.ps1:162]: But this isn't really installing XDP, it's downloading it. And we will be needing this script to actually INSTALL xdp for the xdp tests.

Comment (nibanks) [reply scripts/prepare-machine.ps1:162]: It can be adapted to do that when you onboard XDP tests.

Comment (maolson-msft) [reply scripts/prepare-machine.ps1:162]: I don't get it. This name is misleading. What was wrong with the original name?

Comment (nibanks) [reply scripts/prepare-machine.ps1:162]: When you "install" an SDK, which consists of just headers and libraries used to build, is that not installing in your opinion?

Comment (maolson-msft) [reply scripts/prepare-machine.ps1:162]: I agree that the term "install" is commonly used for SDKs when there is no potential confusion. But that isn't the case here. The XDP SDK can be installed, or the XDP driver can be installed.

It seems to me like your focus is on the name being consistent with other "Install-Xxx" functions. If so, then how about Install-Xdp-Sdk?

Comment (nibanks) [reply scripts/prepare-machine.ps1:162]: Ok. I can do that.

**Review by maolson-msft — APPROVED** [review]

**Review by thhous-msft — APPROVED** [review]

**Pull request merged**